From 1986d4ccffabad84ffa4ea2d2fcbd21423a2f24f Mon Sep 17 00:00:00 2001
From: Balamurugan Selvarajan <quic_bselvara@quicinc.com>
Date: Tue, 8 Mar 2022 18:49:10 +0530
Subject: [PATCH] ath12k: Parse memory-region from reserved memory

parse memory from reserved region. In this case
host need not allocate memory and the memory is
fetched by parsing memory-region property.

Signed-off-by: Balamurugan Selvarajan <quic_bselvara@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/pci.c | 2 +-
 drivers/net/wireless/ath/ath12k/qmi.c | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/net/wireless/ath/ath12k/pci.c b/drivers/net/wireless/ath/ath12k/pci.c
index 31bbcb8..11018fa 100644
--- a/drivers/net/wireless/ath/ath12k/pci.c
+++ b/drivers/net/wireless/ath/ath12k/pci.c
@@ -1303,7 +1303,7 @@ static int ath12k_pci_probe(struct pci_dev *pdev,
 	 * from DT. If memory is reserved from DT for FW, ath11k driver need not
 	 * allocate memory.
 	 */
-	if (!of_property_read_u32(ab->dev->of_node, "ath-memory-region", &addr))
+	if (!of_property_read_u32(ab->dev->of_node, "memory-region", &addr))
 		ab->bus_params.fixed_mem_region = true;
 
 	ret = ath12k_pci_claim(ab_pci, pdev);
diff --git a/drivers/net/wireless/ath/ath12k/qmi.c b/drivers/net/wireless/ath/ath12k/qmi.c
index 250a298..b92fa98 100644
--- a/drivers/net/wireless/ath/ath12k/qmi.c
+++ b/drivers/net/wireless/ath/ath12k/qmi.c
@@ -1951,7 +1951,7 @@ static int ath12k_qmi_assign_target_mem_chunk(struct ath12k_base *ab)
 	for (i = 0, idx = 0; i < ab->qmi.mem_seg_count; i++) {
 		switch (ab->qmi.target_mem[i].type) {
 		case HOST_DDR_REGION_TYPE:
-			hremote_node = of_parse_phandle(dev->of_node, "ath-memory-region", 0);
+			hremote_node = of_parse_phandle(dev->of_node, "memory-region", 0);
 			if (!hremote_node) {
 				ath12k_warn(ab, "qmi fail to get hremote_node\n");
 				return ret;
-- 
2.17.1

